### ISSUE TITLE
Url segments in query string parameters don't work for first parameter

### ISSUE THREAD

**Issue by ashic** [FubarDevelopment/restsharp.portable#16]
Consider the following url:

/api/foo?x={x}&y={y}&z={z}

When adding url segments x, y and z, it seems RestSharp.Portable creates the following url after expansion:

/api/foo??x={x}&y={y}&z={z}

Note the extra question mark. On the server (at least in web api), the first parameter as seen as "?x" instead of x, and as such doesn't bind x.

The following url template works:

/api/foo?&x={x}&y={y}&z={z}

which expands to

/api/foo??&x={x}&y={y}&z={z}

This works as it thinks x is the second parameter, and binds correctly. However that's still a bit ugly.

I checked in RestSharp (i.e. not portable) and it doesn't create the double ? problem. Wondering if this is a difficult fix.

**Comment by sohayb** [comment]
Hi Ashic,
```csharp
            var request = new RestRequest(HttpMethod.Get);
            request.Resource = PATH_HEADER + "/api/foo";
            request.AddParameter("x", xcal, ParameterType.QueryString);
            request.AddParameter("y", yval, ParameterType.QueryString);
            request.AddParameter("z", zval, ParameterType.QueryString);
```
try to execute such request and you'll be just fine, the key is not to include the "?"

**Comment by ashic** [comment]
Hi Sohayb,
I'm aware of that. And that's fine if you're filling in quuery string parameters. However, if you're trying to do url expansion from templated urls provided to you by the server (for example, like http://stateless.co/hal_specification.html ) , then you don't expect to have control of the url. You're meant to only fill in templates. It is fine for url segments to be part of the query string, and as such this scenario should work. It works on RestSharp :)

**Comment by fubar-coder** [comment]
This is indeed something I missed. It'll be fixed in the next version.

**Comment by fubar-coder** [comment]
The bug is fixed in 2.2.0-pre3

**Comment by ashic** [comment]
Awesome. Thanks :)
